src/components/ColumnAnnotationCard.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState } from 'react';
+import {
+  Card,
+  CardHeader,
+  CardContent,
+  Typography,
+  TextField,
+  ToggleButtonGroup,
+  ToggleButton,
+  Fab,
+  Autocomplete,
+} from '@mui/material';
+import EditIcon from '@mui/icons-material/Edit';
+import SaveIcon from '@mui/icons-material/Save';
+import { StandardizedVarible, StandardizedVaribles } from '../utils/types';
+
+interface ColumnAnnotationCardProps {
+  id: number;
+  header: string;
+  description: string | null;
+  dataType: 'Categorical' | 'Continuous' | null;
+  standardizedVariable: StandardizedVarible | null;
+  standardizedVariableOptions: StandardizedVaribles;
+  onDescriptionChange: (columnId: number, newDescription: string | null) => void;
+  onDataTypeChange: (columnId: number, newDataType: 'Categorical' | 'Continuous' | null) => void;
+  onStandardizedVariableChange: (
+    columnId: number,
+    newStandardizedVariable: StandardizedVarible | null
+  ) => void;
+}
+
+function ColumnAnnotationCard({
+  id,
+  header,
+  description,
+  dataType,
+  standardizedVariable,
+  standardizedVariableOptions,
+  onDescriptionChange,
+  onDataTypeChange,
+  onStandardizedVariableChange,
+}: ColumnAnnotationCardProps) {
+  const [isEditingDescription, setIsEditingDescription] = useState(false);
+  const [editedDescription, setEditedDescription] = useState<string | null>(description);
+  const [selectedDataType, setSelectedDataType] = useState<'Categorical' | 'Continuous' | null>(
+    dataType
+  );
+  const [selectedStandardizedVariableKey, setSelectedStandardizedVariableKey] = useState<
+    string | null
+  >(standardizedVariable ? standardizedVariable.identifier : null);
+
+  const handleEditDescription = () => {
+    setIsEditingDescription(true);
+  };
+
+  const handleSaveDescription = () => {
+    setIsEditingDescription(false);
+    onDescriptionChange(id, editedDescription);
+  };
+
+  const handleDataTypeChange = (
+    _: React.MouseEvent<HTMLElement>,
+    newDataType: 'Categorical' | 'Continuous' | null
+  ) => {
+    setSelectedDataType(newDataType);
+    onDataTypeChange(id, newDataType);
+  };
+
+  const handleStandardizedVariableChange = (
+    _: React.ChangeEvent<unknown>,
+    newValue: string | null
+  ) => {
+    setSelectedStandardizedVariableKey(newValue);
```

Comment:
This is incorrect. 1) because you're passing the string "label" but expect a key,  2) and more importantly because you are creating duplicated state here between the component state and the store state (set just 2 lines below)